Sources/PostgresNIO/New/PostgresChannelHandler.swift

```diff
@@ -212,23 +212,27 @@ final class PostgresChannelHandler: ChannelDuplexHandler {
                 psqlTask = self.makeStartListeningQuery(channel: channel, context: context)
 
             case .none:
+                promise?.succeed(Void())
                 return
 
             case .succeedListenStart(let listener):
+                precondition(promise == nil, "This Promise will leak.")
```

Comment:
WDYT about these preconditions?
I tried to maximize the amount of preconditions, so for all the no-precondition paths there is a test that would otherwise trigger the precondition if it was actually there.

Reply:
It also appears these "listen"-related paths are not very well tested. Need to be careful with the preconditions.

Reply:
In this case we are already listening to channel in question. Just succeed the promise here as above.